app/controllers/api/v1/application_controller.rb

```diff
@@ -0,0 +1,7 @@
+class Api::V1::ApplicationController < ActionController::Base
+  include CurrentUser
+
+  def warden_strategies
```

Comment:
Correct!

Regarding the constant I think that overriding a method is somewhat more mainstream way of changing behavior in ruby inheritance (and mixins).
Constants have some weird effects depending on which module you are currently at (this is why we need things like `self.class::CONSTANT`).
There's also things like the ActiveSupport hooks that try to load files to auto detect constants.
In general they also limit the way I can decide what to return to class "compile" time instead of runtime.

I'll make this method private as you suggest.